corehq/apps/saved_reports/models.py

```diff
@@ -667,22 +671,66 @@ def _get_and_send_report(self, language, emails):
             )
 
             attach_excel = getattr(self, 'attach_excel', False)
-            content, excel_files = get_scheduled_report_response(
-                self.owner, self.domain, self._id, attach_excel=attach_excel,
-                send_only_active=True
-            )
+            try:
+                content, excel_files = get_scheduled_report_response(
+                    self.owner, self.domain, self._id, attach_excel=attach_excel,
+                    send_only_active=True
+                )
 
-            # Will be False if ALL the ReportConfigs in the ReportNotification
-            # have a start_date in the future.
-            if content is False:
-                return
-
-            for email in emails:
-                body = render_full_report_notification(None, content, email, self).content
-                send_html_email_async.delay(
-                    title, email, body,
-                    email_from=settings.DEFAULT_FROM_EMAIL,
-                    file_attachments=excel_files)
+                # Will be False if ALL the ReportConfigs in the ReportNotification
+                # have a start_date in the future.
+                if content is False:
+                    return
+
+                for email in emails:
+                    body = render_full_report_notification(None, content, email, self).content
+                    send_html_email_async.delay(
+                        title, email, body,
+                        email_from=settings.DEFAULT_FROM_EMAIL,
+                        file_attachments=excel_files)
+            except Exception as er:
+                notify_exception(
+                    None,
+                    message="Encountered error while generating report or sending email",
+                    details={
+                        'subject': title,
+                        'recipients': str(emails),
+                        'error': er,
+                    }
+                )
+                if getattr(er, 'smtp_code', None) in LARGE_FILE_SIZE_ERROR_CODES or type(er) == ESError:
+                    # If the email doesn't work because it is too large to fit in the HTML body,
+                    # send it as an excel attachment, by creating a mock request with the right data.
+                    mock_request = HttpRequest()
+                    mock_request.couch_user = self.owner
+                    mock_request.user = self.owner.get_django_user()
+                    mock_request.domain = self.domain
+                    mock_request.couch_user.current_domain = self.domain
+                    mock_request.couch_user.language = 'lang (PV - fix this)'
+                    mock_request.method = 'GET'
+                    mock_request.bypass_two_factor = True
+
+                    report_config = self.configs[0]
```

Comment:
I think you want to iterate over configs, not just grab the first one. Scheduled reports can have more than one report in them.

Reply:
https://github.com/dimagi/commcare-hq/pull/24007/commits/18b04eb4a0c316f80754bfbbe258e01cc4bd8982